src/main/java/com/example/onlinebookstore/service/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.example.onlinebookstore.service;
+
+import com.example.onlinebookstore.dto.cartitem.CartItemResponseDto;
+import com.example.onlinebookstore.dto.shoppingcart.ShoppingCartDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.CartItemMapper;
+import com.example.onlinebookstore.mapper.ShoppingCartMapper;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.BookRepository;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.repository.ShoppingCartRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@AllArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+
+    @Override
+    public ShoppingCartDto getCartByUser(User user) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId())
+                .orElseGet(() -> shoppingCartRepository.save(new ShoppingCart(user)));
```

Comment:
we can add some `createShoppingCart()` method and use it here and in other places like `userService.register()`

Reply:
As far as I understood i should create shopping cart when a user registers. But I already have createing a new shopping cart in userService